### PR TITLE
Add node activity health reporters on node manager

### PR DESCRIPTION
**Pull request by derailed**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Add instrumentation to node manager to provide module health reporting when node
CRUD operations are detected.

```release-note
Add node activity health reporters on node manager
```

### CODE REVIEW

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -1636,21 +1630,29 @@ func (n *linuxNodeHandler) replaceNodeIPSecOutRoute(ip *net.IPNet) error {
 	return nil
 }
 
-// replaceNodeExternalIPSecOutRoute replace the out IPSec route in the host
-// routing table with the new route. If no route exists the route is installed
-// on the host. The caller must ensure that the CIDR passed in must be non-nil.
-func (n *linuxNodeHandler) replaceNodeExternalIPSecOutRoute(ip *net.IPNet) error {
-	var errs error
+func isValidIP(cfg datapath.LocalNodeConfiguration, ip *net.IPNet) bool {
```
Comment (danehans) [inline pkg/datapath/linux/node.go]: Please add godocs for this function.

**Review by danehans — CHANGES_REQUESTED** [review]
Thanks for the PR! I have a few minor pieces of feedback:
- Changes are included that are inconsistent with the PR's description. Please use a separate PR for the changes unrelated to adding the health reporter.
- It would be helpful to create and link an issue to provide additional context for the PR.

File: pkg/datapath/linux/node.go (hunk 1 above)
Comment (derailed) [reply pkg/datapath/linux/node.go]: I am not sure this function warrants godocs. It is private and fairly trivial for readers to assert what it does?

**Comment by derailed** [comment]
@danehans Thank you for the review and comments! I'll update accordingly...

**Review by danehans — APPROVED** [review]

**Review by brb — APPROVED** [review]
Thanks!

**Pull request merged**